mmseg/models/backbones/mit.py

```diff
@@ -0,0 +1,454 @@
+            Attention of Segformer. Default: 1.
+    """
+
+    def __init__(self,
+                 embed_dims,
+                 num_heads,
+                 feedforward_channels,
+                 drop_rate=0.,
+                 attn_drop_rate=0.,
+                 drop_path_rate=0.,
+                 num_fcs=2,
+                 qkv_bias=True,
+                 act_cfg=dict(type='GELU'),
+                 norm_cfg=dict(type='LN'),
+                 batch_first=True,
+                 sr_ratio=1):
+        super(TransformerEncoderLayer, self).__init__()
+
+        _, self.norm1 = build_norm_layer(norm_cfg, embed_dims)
+
+        self.attn = EfficientMultiheadAttention(
+            embed_dims=embed_dims,
+            num_heads=num_heads,
+            attn_drop=attn_drop_rate,
+            proj_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            batch_first=batch_first,
+            qkv_bias=qkv_bias,
+            norm_cfg=norm_cfg,
+            sr_ratio=sr_ratio)
+
+        _, self.norm2 = build_norm_layer(norm_cfg, embed_dims)
+
+        self.ffn = MixFFN(
+            embed_dims=embed_dims,
+            feedforward_channels=feedforward_channels,
+            num_fcs=num_fcs,
+            ffn_drop=drop_rate,
+            dropout_layer=dict(type='DropPath', drop_prob=drop_path_rate),
+            act_cfg=act_cfg)
+
+    def forward(self, x, H, W):
+        x = self.attn(self.norm1(x), H, W, identity=x)
+        x = self.ffn(self.norm2(x), H, W, identity=x)
+        return x
+
+
+class OverlapPatchEmbed(BaseModule):
+    """Image to Patch Embedding."""
+
+    def __init__(self,
+                 patch_size,
+                 in_channels,
+                 embed_dims,
+                 stride,
+                 norm_cfg=None):
+        super().__init__()
+
+        self.proj = ConvModule(
+            in_channels=in_channels,
+            out_channels=embed_dims,
+            kernel_size=patch_size,
+            stride=stride,
+            padding=patch_size // 2,
+            act_cfg=None,
+            norm_cfg=None)
+        _, self.norm = build_norm_layer(norm_cfg, embed_dims)
+
+    def forward(self, x):
+        x = self.proj(x)
+        _, _, H, W = x.shape
+        x = nchw_to_nlc(x)
+        x = self.norm(x)
+
+        return x, H, W
+
+
+@BACKBONES.register_module()
+class MixVisionTransformer(BaseModule):
+    """The backbone of Segformer.
+
+    A PyTorch implement of : `SegFormer: Simple and Efficient Design for
+    Semantic Segmentation with Transformers` -
+        https://arxiv.org/pdf/2105.15203.pdf
+    """
+
+    def __init__(self,
+                 in_channels=3,
+                 embed_dims=[64, 128, 256, 512],
+                 num_layers=[3, 4, 6, 3],
+                 num_heads=[1, 2, 4, 8],
+                 mlp_ratios=[4, 4, 4, 4],
+                 out_indices=(0, 1, 2, 3),
+                 qkv_bias=True,
```

Comment:
Add an assert for the length of related `args` (refer to [ResNet](https://github.com/open-mmlab/mmsegmentation/blob/master/mmseg/models/backbones/resnet.py#L447)).